src/mutex/mutex.go

```diff
@@ -4,11 +4,17 @@ import (
 	"sync"
 )
 
-var mutexes = make(map[int64]*sync.Mutex)
+var (
+	mutexes = make(map[int64]*sync.Mutex)
+	lock    = new(sync.Mutex)
+)
 
 // Lock locks the mutex for a given proposal, after instanciating if it doesn't
 // exist.
 func Lock(number int64) {
+	// Avoid multiple goroutines using the map at the same time.
+	lock.Lock()
+
 	// Lock the existing mutex if there's one.
 	if m, exists := mutexes[number]; exists {
 		m.Lock()
```

Comment:
You never unlock `lock` if `mutexes[number]` exists

Reply:
Addressed by deferring call to Unlock() rather than having it happen at the end of the function.